python/cudf/dataframe/dataframe.py

```diff
@@ -2663,6 +2662,7 @@ def from_pandas(cls, dataframe, nan_as_null=True):
                     for idx in range(len(vals.shape)):
                         df[colk+str(idx)] = Series(vals[idx],
                                                    nan_as_null=nan_as_null)
+        df.columns = dataframe.columns
```

Comment:
What's the specific reason for this line? If the source dataframe doesn't have multiindexed columns, then `df.columns` will already match `dataframe.columns`. If the source dataframe does have multiindex columns, then they will be setdown on 2675.

Reply:
Repeated this, thought I'd lost my review.